api/tests/routes/native/v1/account_test.py

```diff
@@ -868,6 +868,107 @@ def test_ongoing_email_update_blocks_the_next_update_request(self, client):
+        base_url_validation = parse_qs(validation_link.query)["link"][0]
+        base_url_params_validation = parse_qs(urlparse(base_url_validation).query)
+        assert {"new_email", "token", "expiration_timestamp"} <= base_url_params_validation.keys()
+        assert base_url_params_validation["new_email"] == [new_email]
+        user = users_models.User.query.get(user.id)
+        assert user.email == old_email
+        response = client.put(
+            "/native/v1/profile/email_update/validate",
+            json={
+                "token": base_url_params_validation["token"][0],
+            },
+        )
+        assert response.status_code == 204
+        user = users_models.User.query.get(user.id)
+        assert user.email == new_email
+        assert len(mails_testing.outbox) == 3
+
+    def test_end_to_end_update_email_with_cancellation(self, client):
+        password = "some_random_string"
+        old_email = self.identifier
+        new_email = "updated_" + old_email
+        user = users_factories.UserFactory(email=old_email, password=password)
+        client.with_token(user.email)
+        response = client.post(
+            "/native/v1/profile/update_email",
+            json={
+                "email": new_email,
+                "password": password,
+            },
+        )
+        assert response.status_code == 204
+        assert mails_testing.outbox
+
+        confirmation_email = mails_testing.outbox[-1]
+        confirmation_link = urlparse(confirmation_email.sent_data["params"]["CONFIRMATION_LINK"])
+        cancellation_link = urlparse(confirmation_email.sent_data["params"]["CANCELLATION_LINK"])
+        base_url_confirmation = parse_qs(confirmation_link.query)["link"][0]
+        base_url_cancellation = parse_qs(cancellation_link.query)["link"][0]
+        base_url_params_confirmation = parse_qs(urlparse(base_url_confirmation).query)
+        base_url_params_cancellation = parse_qs(urlparse(base_url_cancellation).query)
+        assert {"new_email", "token", "expiration_timestamp"} <= base_url_params_confirmation.keys()
+        assert {"new_email", "token", "expiration_timestamp"} <= base_url_params_cancellation.keys()
+        assert base_url_params_confirmation["new_email"] == [new_email]
+        assert base_url_params_cancellation["new_email"] == [new_email]
+        assert base_url_params_cancellation["token"] == base_url_params_confirmation["token"]
+        user = users_models.User.query.get(user.id)
+        assert user.email == old_email
+
+        response = client.post(
+            "/native/v1/profile/email_update/cancel",
+            json={
+                "token": base_url_params_confirmation["token"][0],
+            },
+        )
+        assert response.status_code == 204
+        user = users_models.User.query.get(user.id)
+        assert user.email == old_email
+        assert user.isActive == False
```

Comment:
(style) Plus "pythonique" : 

```suggestion
        assert not user.isActive
```